src/blockrelay/graphene_set.h

```diff
@@ -128,12 +132,24 @@ class CGrapheneSet
 
     static std::vector<uint64_t> DecodeRank(std::vector<unsigned char> encoded, size_t nItems, uint16_t nBitsPerItem);
 
-    uint64_t GetFilterSerializationSize() { return ::GetSerializeSize(*pSetFilter, SER_NETWORK, PROTOCOL_VERSION); }
+    uint64_t GetFilterSerializationSize()
+    {
+        if (computeOptimized)
+            return ::GetSerializeSize(*pFastFilter, SER_NETWORK, PROTOCOL_VERSION);
+        else
+            return ::GetSerializeSize(*pSetFilter, SER_NETWORK, PROTOCOL_VERSION);
+    }
     uint64_t GetIbltSerializationSize() { return ::GetSerializeSize(*pSetIblt, SER_NETWORK, PROTOCOL_VERSION); }
     uint64_t GetRankSerializationSize() { return ::GetSerializeSize(encodedRank, SER_NETWORK, PROTOCOL_VERSION); }
     ~CGrapheneSet()
     {
-        if (pSetFilter)
+        if (computeOptimized && pFastFilter)
```

Comment:
from a defensive programming standpoint you want to delete these objects if they were allocated, no matter the value of computeOptimized.  So I would prefer just "if (pFastFilter)".  This change would have no semantic difference right now because ((bool)computeOptimized) implies ((bool)pFastFilter).  But cleaning these up no matter what may avoid a memory leak if someone modifies things in the future.

Reply:
Done.